### PR TITLE
Lfs config

### PR DESCRIPTION
**Pull request by cpt-davy-jones**
Added possibility of using tag <LfsUrl>Https://Some.Url.com/</Url> to create the .lfsconfig file containing the given Url. 
Will not overwrite existing .lfsconfig if it was already added to Starteam.

### CODE REVIEW

File: syncronizer/src/org/ossnoize/git/fastimport/Commit.java [hunk 1]
```diff
@@ -107,6 +118,26 @@ public void writeTo(OutputStream out) throws IOException {
...
+				String lfsString = "[lfs]\n";
+				lfsconfigFile.writeData(lfsString.getBytes("UTF-8"));
+				String urlString = "    url = " + lfsConfigUrl;
+				lfsconfigFile.writeData(urlString.getBytes("UTF-8"));
+				Blob aLfsMarkedBlob = new Blob(lfsconfigFile);
+				aLfsMarkedBlob.writeTo(out);
+				FileModification lfsconfig = new FileModification(aLfsMarkedBlob);
+				lfsconfig.setFileType(GitFileType.Normal);
+				lfsconfig.setPath(".lfsconfig");
+				this.addFileOperation(lfsconfig);
```
Comment (planestraveler) [inline syncronizer/src/org/ossnoize/git/fastimport/Commit.java]: I would have preferred an actual Object managing the `.lfsconfig` file. Enabling further extension afterward in the same spirit as the .`gitattribute`

File: syncronizer/src/org/sync/GitImporter.java [hunk 2]
```diff
@@ -388,6 +392,9 @@ else if(StarteamEOL.LF.value() == (Integer) propertyValue){
 					if (fattributes != null) {
 						commit.setAttributes(fattributes);
 					}
+					if(lfsConfigUrl != null){
+						commit.setLfsConfigUrl(lfsConfigUrl);
+					}
```
Comment (planestraveler) [inline syncronizer/src/org/sync/GitImporter.java]: To what I see here, you add the `.lfsconfig` to each commit. This isn't really desirable for everyone.

File: syncronizer/src/org/ossnoize/git/fastimport/Commit.java [hunk 3]
```diff
@@ -107,6 +118,26 @@ public void writeTo(OutputStream out) throws IOException {
 				attributes.setFileType(GitFileType.Normal);
 				attributes.setPath(".gitattributes");
 				this.addFileOperation(attributes);
+
+			} catch (InvalidPathException ex) {
+			}
+		}
+
+		Set<String> listOfTrackedFiles = repositoryHelper.getListOfTrackedFile(reference);
```
Comment (planestraveler) [inline syncronizer/src/org/ossnoize/git/fastimport/Commit.java]: I don't think this is the best place to check if the file already exist. The commit object should only take care of committing files and nothing else. This check should be done before and the decision to add it or not before. Generating the file at this point is correct.

File: syncronizer/src/org/sync/GitImporter.java [hunk 4]
```diff
@@ -374,6 +375,9 @@ else if(StarteamEOL.LF.value() == (Integer) propertyValue){
 					if (fattributes != null) {
 						lastCommit.setAttributes(fattributes);
 					}
+					if(lfsConfigUrl != null){
+						lastCommit.setLfsConfigUrl(lfsConfigUrl);
+					}
```
Comment (planestraveler) [inline syncronizer/src/org/sync/GitImporter.java]: When we start a migration that is good, but somehow if the file already exist in the repository. Should we keep it as is or should we overwrite it? What if some other parameters were already configured but without the URL?

Comment (cpt-davy-jones) [reply syncronizer/src/org/sync/GitImporter.java]: For now, if the .lfsconfig file already exist in the repository. We keep it as is.

**Pull request merged**